modules/spec/unit/modules/puppet_spec.rb

```diff
@@ -170,6 +201,7 @@
            "transaction_uuid" => "ac59acbe-6a0f-49c9-8ece-f781a689fda9",
            "environment"      => "production",
            "status"           => "changed",
+           "error_type"       => "",
```

Comment:
I would prefer it if these fields weren't present on a successful run. This could however make it harder for NUI and App Mgmt to integrate these changes. I suggest we keep the behavior as it for now, run the change past them and then make "error" and "error_type" optional in another commit.